docs/deployment/overview.md

```diff
@@ -1,47 +1,301 @@
 
-For more information on the schema of the manifest and how to run AppHost project targets, see [.NET Aspire manifest format for deployment tool builders](manifest-format.md).
+The `aspire publish` command produces deployment artifacts that contain unresolved parameters (placeholders). The `aspire deploy` command uses these artifacts, resolves the parameters when supported by the target integration, and then executes the deployment. Some integrations don't support the `deploy` command.
 
-## Deploy to Azure
+## Aspire CLI commands (conceptual behavior)
 
-.NET Aspire enables deployment to Azure Container Apps or Azure App Service when using the ASpire CLI. The number of environments .NET Aspire can deploy to will grow over time.
+| Command | What it does | Outputs | Parameter state | Requires integration capability |
+|---------|--------------|---------|-----------------|---------------------------------|
+| `aspire publish` | Transforms the application model into integration-specific assets (Compose files, manifests, specifications, etc.). | Intermediate artifacts (not directly production-final). | Unresolved (placeholders, e.g. `${VAR}` or similar). | Publish support |
+| `aspire deploy` | Runs a deployment using one or more integrations (build, parameter resolution, apply). | Real resources / applied changes. | Resolved. | Deploy support |
 
-### Azure Container Apps
+If an integration does not implement deploy functionality, `aspire deploy` will not deploy that target (it may warn or no-op for it).
 
-.NET Aspire projects are designed to run in containerized environments. Azure Container Apps is a fully managed environment that enables you to run microservices and containerized applications on a serverless platform. The [Azure Container Apps](azure/aca-deployment.md) topic describes how to deploy Aspire apps to ACA using the Aspire CLI or Azure Developer CLI (azd).
+When you run `aspire deploy` without any integrations that support deployment, you'll see this error:
 
-### Use Application Insights for .NET Aspire telemetry
+```Output
+FAILED: Analyzing the distributed application model for publishing and deployment capabilities.
+           No resources in the distributed application model support deployment.
+```
 
-.NET Aspire projects are designed to emit telemetry using OpenTelemetry which uses a provider model. .NET Aspire projects can direct their telemetry to Azure Monitor / Application Insights using the Azure Monitor telemetry distro. For more information, see [Use Application Insights for .NET Aspire telemetry](azure/application-insights.md) for step-by-step instructions.
+Similarly, when you run `aspire publish` without any integrations that support publishing, you'll see:
 
-## Deploy to Kubernetes
+```Output
+Analyzing the distributed application model for publishing and deployment capabilities. 00:00:00
+           No resources in the distributed application model support publishing.
+```
```

Comment:
I'd change the order here to show `aspire `publish` first, since it occurs first logically in all scenarios (I'm actually curious is it should be considered a prerequisite to `aspire deploy` since you must call it first in all cases that I'm familiar with. I cannot think of a scenario where you wouldn't call `aspire publish`, but we might want to call that out too).